File: MonetDB/src/mapi/clients/MapiClient.py
from Mapi import * 
import fileinput;

s = server( hostname(), portnr(), environ['USER'] + '\n')
fi = fileinput.FileInput()
print("#MapiClient (python) connected to %s:%d as %s" % (hostname(), portnr(), environ['USER']))
sys.stdout.write( s.prompt )
line= fi.readline()
while( line != "quit;\n" ):
	res = s.cmd( line )
	print(res);
	sys.stdout.write( s.prompt )
	line = fi.readline() 
s.disconnect()

File: MonetDB/src/mapi/clients/MapiClient.pl
#!/usr/bin/perl

use Mapi;

my ($monet, $line);

$monet = new Mapi( "localhost:50000",  $ENV{'USER'});
#$monet = new Mapi( $ENV{'MAPIPORT'},  $ENV{'USER'});

print ">";
while ( !(($line=<>) =~ /quit;/) ){
	print $monet->cmd($line);
	print ">";
}

$monet->disconnect();

1;

